### DISCUSSION TITLE
Score multiplier hangs at last digit slightly too long when disabling mods.

### DISCUSSION THREAD

**Discussion by Stoppedpuma** [ppy/osu#26553] [Q&A] [answered]
I don't know if this is intentional or not but it looks odd to my eyes.

https://github.com/ppy/osu/assets/58333920/9a49d360-8dbf-4cc4-82f8-af627b323789

**Comment by sz-da** [comment]
Also noticed this for acc, especially with 100% acc

https://github.com/ppy/osu/assets/76718358/61f8734e-0747-4aa6-beed-12ed5f48e176

**Comment by peppy** [comment]
This is an interesting one I've been conscious of for some time when it comes to animations, but I guess it's much more noticeable when used on numbers. Basically, we (I) like the `OutQuint` easing style, but it sticks short of the final goal for a considerable amount of time.

I'd really like for it to just round when it gets close enough. This would improve performance game-wide as we would have tweens/invalidations performing invisible changes. @bdach what are your thoughts on this? Would changing how `OutQuint` works at a framework level be seen as a "breaking change" or "fixing a bug"? I'd argue that anyone using this would see it as a welcome change.

If that's off the table, then we could add a new easing type and change all osu usages to it.

**Reply by bdach** [reply]
To begin with I'm confused as to what's being proposed here. What do you want to do to `OutQuint`? Change it in general to a different formula? Only attempt to patch this individual case somehow?

That said no interpretation of the proposal sounds like a good idea to me. `OutQuint` is named after the fifth power, so if it were to become *something else*, that'd be a misnomer. And also I believe that easing exists in stable, so changing it would no longer match (unsure if that actually matters anywhere but yeah).

I'd say custom easing is preferable at this stage.

**Reply by peppy** [reply]
Yeah, proposal is a different formula which doesn't stick at the end for 40% of the transition's time.

Something like `OutQuintFast` or `OutQuintClipped`?

**Reply by bdach** [reply]
I mean if that is the goal then I'd say the easiest solution seems to be to either apply harsher powers (e.g. `OutExpo`), or _less_ harsh powers (i.e. `OutQuad`) _and_ decrease durations.

**Reply by peppy** [reply]
Maybe, but I'd have to recalibrate my expectations of what numbers map to what tween 😅 . It'd also be a bit of a pain to update basically everywhere in osu!.

I can agree that for this isolated usage just changing it makes sense though (okay on checking it's still quite a few usages to be checked).

**Reply by bdach** [reply]
I'd say this only realistically matters for rolling counters or other usages that expose raw values somehow. Elsewhere nobody is going to notice that the easing does this.

**Reply by peppy** [reply]
> Elsewhere nobody is going to notice that the easing does this.

Except me, when looking into performance issues. The thing is that as I mentioned earlier, it triggers invalidation and transform overheads with nothing actually changing on screen. And worse case, you get a tiny subpixel movement way after you expect a transform to have finished..

**Reply by bdach** [reply]
> it triggers invalidation and transform overheads with nothing actually changing on screen

Well if that is also the concern then the fix is to just decrease transform length period. Very few places will stop invalidating unless durations are tweaked because I don't recall very many places doing debouncing based on precision checks or the like. And the transform overhead is more associated with the *duration* of the transform than the easing of it.

**Reply by peppy** [reply]
Fair point. I guess what remains is to figure out which other easing type gives similar results with reduced duration.

**Reply by bdach** [reply]
To back this up, here's a graph which intends to show that any `OutQuint` transform should be replaceable by an `OutQuad` transform that's half as long,  or an `OutCubic` transform that's two-thirds as long:

https://www.desmos.com/calculator/irtnszmaai

![1705482105](https://github.com/ppy/osu/assets/20418176/12453dac-d255-4e68-80d2-b19542640002)

**Reply by peppy** [reply, marked as answer]
https://github.com/ppy/osu/pull/26593
